static_analyzer/engine/lsp_client.py

```
@@ -0,0 +1,637 @@
+
+    def __enter__(self) -> LSPClient:
+        self.start()
+        return self
+
+    def __exit__(self, _exc_type: type | None, _exc_val: Exception | None, _exc_tb: object | None) -> None:
+        self.shutdown()
+
+    def start(self) -> dict | list | None:
+        """Start the LSP server process and perform initialization handshake."""
+        env = os.environ.copy()
+        self._process = subprocess.Popen(
```

Ruff (0.15.6)

[error] `subprocess` call: check for execution of untrusted input

(S603)